diff -uNr rootfs.original/home/lava/klipper/klippy/extras/filament_detect.py rootfs/home/lava/klipper/klippy/extras/filament_detect.py
--- rootfs.original/home/lava/klipper/klippy/extras/filament_detect.py	2025-11-06 05:01:19.000000000 +0100
+++ rootfs/home/lava/klipper/klippy/extras/filament_detect.py	2025-12-01 23:11:18.373585258 +0100
@@ -1,5 +1,6 @@
 import logging, copy, os
 from . import filament_protocol
+from . import filament_protocol_ndef
 from . import fm175xx_reader
 from . import filament_feed
 
@@ -124,6 +125,14 @@
                     filament_info = info
                 else:
                     logging.error("channel[%d] m1 parse err: %d", channel, error)
+            elif fm175xx_reader.FM175XX_MIFARE_CARD_TYPE_NTAG == card_type and fm175xx_reader.FM175XX_OK == result:
+                logging.info("channel[%d] trying NDEF parsing....", channel)
+                error, info = filament_protocol_ndef.ndef_proto_data_parse(card_data)
+                if (error == filament_protocol.FILAMENT_PROTO_OK):
+                    logging.info("channel[%d] NDEF parse ok....", channel)
+                    filament_info = info
+                else:
+                    logging.error("channel[%d] NDEF parse err....", channel)
         else:
             is_clear = True
 
